### PR TITLE
UCP/WIREUP: Use UCP_MAX_MDS as md index for CM lane selection

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Use `UCP_MAX_MDS` constant as md index for CM lane selection

## Why ?

It has to be `UCP_MAX_MDS` instead of `UCP_NULL_RESOURCE`
https://github.com/openucx/ucx/blob/a06839401fe3620c8ccd5f9c1b4cbc362d3025b2/src/ucp/core/ucp_ep.c#L71

## How ?

`select_info.rsc_index`->`UCP_MAX_MDS` passing to `ucp_wireup_add_lane_desc`

### CODE REVIEW

**Comment by dmitrygx** [comment]
@alinask @evgeny-leksikov @brminich could you review pls?

File: src/ucp/wireup/select.c [hunk 1]
```diff
@@ -833,7 +833,7 @@ ucp_wireup_add_cm_lane(const ucp_wireup_select_params_t *select_params,
     select_info.path_index = 0;  /**< Only one lane per CM device */
 
     /* server is not a proxy because it can create all lanes connected */
-    return ucp_wireup_add_lane_desc(&select_info, select_info.rsc_index,
+    return ucp_wireup_add_lane_desc(&select_info, UCP_MAX_MDS,
```
Comment (alinask) [inline src/ucp/wireup/select.c:836]: how did it work so far then? were the values the same?

Comment (evgeny-leksikov) [reply src/ucp/wireup/select.c:836]: nope, values are different but seems like the following check is always false by some reason:
```
        if ((lane_desc->rsc_index == select_info->rsc_index) &&
            (lane_desc->addr_index == select_info->addr_index) &&
            (lane_desc->path_index == select_info->path_index))
        {
            lane = lane_desc - select_ctx->lane_descs;
            ucs_assertv_always(dst_md_index == lane_desc->dst_md_index,
                               "lane[%d].dst_md_index=%d, dst_md_index=%d",
                               lane, lane_desc->dst_md_index, dst_md_index);
```

Comment (yosefe) [reply src/ucp/wireup/select.c:836]: we should pass UCP_NULL_RESOURCE here
(which was the case for select_info.rsc_index which was set to UCP_NULL_RESOURCE for CM anyways)

Comment (yosefe) [reply src/ucp/wireup/select.c:836]: and config_key_reset() should be fixed to set NULL_RESOURCE as well

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Review by evgeny-leksikov — APPROVED** [review]

**Pull request merged**